lib/firenest/pub_sub.ex

```diff
@@ -108,13 +108,13 @@ defmodule Firenest.PubSub do
     supervisor = Module.concat(pubsub, "Supervisor")
     dispatcher = Module.concat(pubsub, "Dispatcher")
     registry = [meta: [pubsub: {topology, dispatcher, module, function}],
-                partitions: partitions]
-
-    import Supervisor.Spec
+                partitions: partitions,
+                keys: :duplicate,
+                name: pubsub]
 
     children = [
-      supervisor(Registry, [:duplicate, pubsub, registry]),
-      worker(Dispatcher, [dispatcher, pubsub, module, function])
+      {Registry, registry},
+      {Dispatcher, [dispatcher, pubsub, module, function]}
```

Comment:
Because the dispatcher is private API, we don't need to pass a keyword list, passing a single element is fine, but I would use a tuple. :)

Reply:
Just to be clear, you're suggesting updating to something like `name: dispatcher, opts: {pubsub, module, function}` or equivalent, right?

Reply:
For the Dispatcher in particular I was suggesting to pass a tuple instead of a list and that's it. We don't need keyword args because it is private.